build_platform.py

```diff
@@ -104,7 +114,7 @@ def manually_install_esp32_bsp(repo_info):
     # Assemble git url
     repo_url = "git clone -b {0} https://github.com/{1}/arduino-esp32.git esp32".format(repo_info.split("/")[1], repo_info.split("/")[0])
     # Locally clone repo (https://docs.espressif.com/projects/arduino-esp32/en/latest/installing.html#linux)
-    os.system("mkdir -p /home/runner/Arduino/hardware/espressif")
+    subprocess.run("mkdir -p /home/runner/Arduino/hardware/espressif", shell=True, check=True)
```

Comment:
Why is this being changed from `os.system`?

Reply:
Also, check the subprocess.run result (result = subproces.run(..)) after this operation on L117. You do this a in the calls below.

Reply:
One of my laptops listed it as a problem to use os.system over subprocess module instead. It's not such a concern in this situation, but generally there are problems with it returning too soon and with the codes/exceptions raised not allowing easy determination between a bad process's execution result or the spawning of the process failed / some other error. There's also stuff related to argument lists (string array for non shell) and shells. https://github.com/python/cpython/issues/120743

I hadn't spotted it was using check=True which raises an exception if non-zero result code, but it was a good catch and I'm not sure what the behaviour would have been. I've adjusted it to check and print a nicer (more informative) result and exit with -1